src/DataStructures/DataBox.hpp

```diff
@@ -553,9 +553,9 @@ struct create_dependency_graph {
 template <typename Caller, typename Callee, typename List>
 struct create_dependency_graph<
     Caller, Callee, List, Requires<is_simple_compute_item<Callee>::value>> {
-  using sub_tree =
-      tmpl::fold<typename Callee::argument_tags, List,
-                 create_dependency_graph<Callee, tmpl::_element, tmpl::_state>>;
+  using sub_tree = tmpl::fold<
```

Comment:
Please add a better commit message, as we had no initial idea what this commit is doing

Reply:
Looking over this, I should have made 3 separate commits, I will refactor it so that it's clear what's happening. The commits will be:

1. compute_databox_type metafunction
2. fix mutate bug (resetting of Variables-compute items did not work correctly, leading to confusing compiler errors at best)
3. mutate_apply function

Reply:
👍